hack/generator/pkg/codegen/pipeline_delete_generated_code.go

```diff
@@ -0,0 +1,162 @@
+package codegen
+
+import (
+	"bufio"
+	"context"
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+	"github.com/bmatcuk/doublestar"
+	"github.com/pkg/errors"
+	"io"
+	"io/ioutil"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+// deleteGeneratedCode creates a pipeline stage for cleanup of our output folder prior to generating files
+func deleteGeneratedCode(ctx context.Context, outputFolder string) PipelineStage {
```

Comment:
I like that this code was moved to another file (as it was cluttering up code_gen), but I'm not sure that I like this as a pipeline stage.

On the one hand, it works just fine.
On the other hand, it doesn't actually do any transformation on the type collection (`types`), nor does it actually use `types` at all, which is sorta a red flag to me - This function is just one really just one big side-effect.

I was thinking of the pipeline as this sorta functional-programming style pipeline of pure functions (which `exportFilters` is, and I think @babbageclunk's PR would be too), and this violates that purity.

Reply:
Put maybe more succinctly, I was thinking about the pipeline as operating on `types`, like you have it here, and cleaning up the filesystem doesn't operate on types, it operates on the file system.

Reply:
I confess that this is a bit klugy because it doesn't actually change the set of types. Partly did this to have more than one pipeline stage (as a demo), partly because moving the code out of `code_generator.go` made sense, and partly to defer cleanup until just prior to writing the new files.